pypeit/core/flat.py

```diff
@@ -1,486 +1,416 @@
+            determine the slit illumination function, and thus sets
+            the minimum scale on which the illumination function will
+            have features.
+        illum_iter (:obj:`int`, optional):
+            Iteratively construct the slit illumination profile and
+            reject outliers. To include rejection iterations, this
+            must be larger than 0, and you have to provide the sigma
+            threshold (``illum_rej``); otherwise, no iterations are
+            performed.
+        illum_rej (:obj:`float`, optional):
+            Sigma rejection threshold for iterations. If None, no
+            rejection iterations will be performed, regardless of the
+            value of ``illum_iter``.
+        debug (:obj:`bool`, optional):
+            Construct plots output to the screen that show the result
+            of each iteration. Regardless of this flag, no plots are
+            shown if there are no iterations.
+
+    Returns:
+        tuple: Five `numpy.ndarray`_ objects are returned:
+
+            - A boolean array with the pixels used in the
+              construction of the illumination profile. Shape is
+              identical to ``norm_spec``.
+            - A vector with the length of the number of good pixels
+              (sum of the first returned object) with the indices
+              that sorts the flattened list of good coordinate
+              values.
+            - A vector with the sorted coordinate data.
+            - A vector with the data sorted by the respective
+              coordinates.
+            - A vector with the slit illumination profile.
+
+        To construct the empirical 2D illumination profile::
+
+            illum = np.zeros(norm_spec.shape, dtype=float)
+            illum[_spat_gpm] = profile[np.argsort(srt)]
+
+        where ``norm_spec`` is the input array, ``_spat_gpm`` is the
+        first returned object, ``srt`` is the second returned object,
+        and ``profile`` is the last returned object.
+            
     """
+    if illum_rej is None and illum_iter > 0:
+        msgs.warn('Cannot use iterative rejection to construct the illumination function if the '
+                  'rejection is not provided.  Continuing without iteration.')
+
+    _spat_gpm = np.ones(norm_spec.shape, dtype=bool) if spat_gpm is None else np.copy(spat_gpm)
+    _spat_gpm, spat_srt, spat_coo_data, spat_flat_data_raw \
+            = sorted_flat_data(norm_spec, spat_coo, gpm=_spat_gpm)
+    spat_gpm_data_raw = np.ones(spat_flat_data_raw.size, dtype=bool)
+
+    # Assume the density of samples in any given spatial coordinate is
+    # roughly the same at all spatial positions. Calculate the fraction
+    # of the slit width for the median filter as set by the
+    # ``spat_samp`` parameter.
+    med_width = int(np.ceil(np.sum(spat_gpm) * spat_samp / slitwidth))
+
+    # Construct the filtered illumination profile (iteratively if requested)
+    for i in range(illum_iter+1):
```

Comment:
We talked through this.  This isn't because of the bspline fit.  This is an attempt to remove the influence of aberrant points from the construction of the filtered illumination profile (i.e., the data that is fit by the bspline).  By default, this rejection isn't done, and I added a warning to the description of the `illum_rej` flat-field parameter that says this bit of functionality is still being tested.